integration/fuse/src/main/java/alluxio/fuse/AlluxioFuse.java

```diff
@@ -41,51 +47,107 @@
    *
    * @param args arguments to run the command line
    */
   public static void main(String[] args) {
-    AlluxioConfiguration conf = new InstancedConfiguration(ConfigurationUtils.defaults());
+    LOG.info("Alluxio version: {}-{}", RuntimeConstants.VERSION, ProjectConstants.REVISION);
+    AlluxioConfiguration conf = InstancedConfiguration.defaults();
+    FileSystemContext fsContext = FileSystemContext.create(conf);
+    try {
+      InetSocketAddress confMasterAddress =
+          fsContext.getMasterClientContext().getConfMasterInquireClient().getPrimaryRpcAddress();
+      RetryUtils.retry("load cluster default configuration with master " + confMasterAddress,
+          () -> fsContext.getClientContext().loadConfIfNotLoaded(confMasterAddress),
+          RetryUtils.defaultClientRetry(
+              conf.getDuration(PropertyKey.USER_RPC_RETRY_MAX_DURATION),
+              conf.getDuration(PropertyKey.USER_RPC_RETRY_BASE_SLEEP_MS),
+              conf.getDuration(PropertyKey.USER_RPC_RETRY_MAX_SLEEP_MS)));
+    } catch (IOException e) {
+      LOG.warn("Failed to load cluster default configuration for Fuse process. "
+          + "Proceed with local configuration for FUSE: {}", e.toString());
+    }
+    conf = fsContext.getClusterConf();
     final AlluxioFuseOptions opts = parseOptions(args, conf);
     if (opts == null) {
       System.exit(1);
     }
-
-    final FileSystem tfs = FileSystem.Factory.create(conf);
-    final AlluxioFuseFileSystem fs = new AlluxioFuseFileSystem(tfs, opts, conf);
-    final List<String> fuseOpts = opts.getFuseOpts();
-    // Force direct_io in FUSE: writes and reads bypass the kernel page
-    // cache and go directly to alluxio. This avoids extra memory copies
-    // in the write path.
-    fuseOpts.add("-odirect_io");
-
-    try {
-      fs.mount(Paths.get(opts.getMountPoint()), true, opts.isDebug(),
-          fuseOpts.toArray(new String[0]));
-      LOG.info("Mounted Alluxio: mount point=\"{}\", opts=\"{}\"",
-          opts.getMountPoint(), fuseOpts.toArray(new String[0]));
-    } catch (FuseException e) {
-      LOG.error("Failed to mount {}", opts.getMountPoint(), e);
-      // only try to umount file system when exception occurred.
-      // jnr-fuse registers JVM shutdown hook to ensure fs.umount()
-      // will be executed when this process is exiting.
-      fs.umount();
-    } finally {
-      try {
-        tfs.close();
-        LOG.info("Closed Alluxio file system.");
-      } catch (Exception e) {
-        LOG.error("Failed to close Alluxio file system", e);
+    try (final FileSystem fs = FileSystem.Factory.create(fsContext)) {
+      final List<String> fuseOpts = opts.getFuseOpts();
+      // Force direct_io in FUSE: writes and reads bypass the kernel page
+      // cache and go directly to alluxio. This avoids extra memory copies
+      // in the write path.
+      // TODO(binfan): support kernel_cache (issues#10840)
+      //fuseOpts.add("-odirect_io");
```

Comment:
If this is commented out, does this mean the comment about `Force direct_io in FUSE...` is inaccurate?

Reply:
updated. it should be in JNR fuse